praxis-isncsci/ui
# Update extra input controls for external incoming exam data

Closes #163 

## Problem

When exam data is being passed by the `ExternalMessagePortProvider`, the system is not updating the extra input controls with the incoming data. 

<img alt="Controls that need updating" width="1040" src="https://github.com/praxis-isncsci/ui/assets/1294355/c383f52a-794c-4bc5-94f3-11383ec9fe01" />

## Solution

- Updated the helper methods that bind the incoming data to the grid cells
- Added entries for each observation to capture the **reason for impairment not due to SCI** and their **specify** fields
- Updated the helper methods that bind the state data to the data to be passed to the external listener when the calculate button is pressed
- Combined the provider methods that set the active cell and the selected cells so there is only one refresh in the UI

## Testing

- [ ] 1. All interface fields are updated when external data comes in

<details>
  <summary>Test case</summary>

  1. Navigate to the [external message provider story](https://64f8d7c6e093108e99084a70-rctzmckabm.chromatic.com/?path=/story/app-externalmessageportprovider--primary)
  2. Press the `Load random exam` button
  3. Confirm that all input fields (vac, dap, lowest non-key muscles) have been updated
  4. Select a cell marked with `*`
  5. Confirm the star flag related fields got set

</details>

- [ ] 2. All data is passed to the external listener when the calculate button is pressed

<details>
  <summary>Test case</summary>

  1. Navigate to the [external message provider story](https://64f8d7c6e093108e99084a70-rctzmckabm.chromatic.com/?path=/story/app-externalmessageportprovider--primary)
  2. Set the values for an exam
  3. Open the developer console
  4. Press the `Calculate` button
  5. Confirm the system prints to the console the object being passed back to the external listener
  6. Confirm all values were passed in the call

</details>
